### PR TITLE
Allow files generated by the buildscripts up to 8Mb in size

### LINKED ISSUE #6134: No hinting for imported library with generated bindings.
## Environment

* **IntelliJ Rust plugin version:**
* **Rust toolchain version:**
* **IDE name and version:** CL-202.6948.80
* **Operating system:** OS Windows 10(amd64) v10.0

## Problem description

I am using the library `godot-rust` and while everything runs fine, there is no intellisense happening for the (generated) types provided by the package.

This can reproduced by cloning my example repository here:
https://github.com/Sibz/godotrust-bugdemo
This may need LLVM installed in order to compile.

On line 15 I have picked an example method that should hint, but does not.
The description for this struct is here: https://docs.rs/gdnative/0.9.0-preview.0/gdnative/api/rigid_body/struct.RigidBody.html

I have contacted the developer and he is aware of the issue but believes the issue is with the code analyzer - it should be able to pick up the bindings generated but it doesn't.

While this is probably an issue with rust-analyzer, I am not sure if CLion uses that for it's intellisense, hence I am reporting the bug here too.
I have reported this on the rust-analyzer repository
https://github.com/rust-analyzer/rust-analyzer/issues/5992

The code compiles and runs fine.

Submitted to CLion bug tracker: https://youtrack.jetbrains.com/issue/CPP-22114

### LINKED ISSUE #8201: Jump to Definition and Code Completion not working for aws_sdk_ec2::Client
## Environment

* **IntelliJ Rust plugin version:**

0.4.160.4261-212

* **Rust toolchain version:**

1.56.0

* **IDE name and version:**

CLion 2021.2.1
Build #CL-212.5080.54, built on August 24, 2021
Runtime version: 11.0.11+9-b1504.16 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 10.15.7
GC: G1 Young Generation, G1 Old Generation
Memory: 6144M
Cores: 16

* **Operating system:**

macOS 10.15.7

## Problem description

When using a [specific dependency](https://crates.io/crates/aws-sdk-ec2/0.0.26-alpha), I am unable to
- jump to definition
- use code completion for certain types provided by the library

## Steps to reproduce

Cargo.toml

```toml
[package]
name = "untitled25"
version = "0.1.0"
edition = "2021"

[dependencies]
aws-config = "0.0.26-alpha"
aws-types = "0.0.26-alpha"
aws-sdk-ec2 =  "0.0.26-alpha"
```

main.rs

```rust
use aws_config::profile::ProfileFileCredentialsProvider;
use aws_types::config::Config;
use aws_types::credentials::SharedCredentialsProvider;
use aws_types::region::Region;

fn main() {
    let region = Region::new("us-west-2".to_owned());
    let profile = SharedCredentialsProvider::new(
        ProfileFileCredentialsProvider::builder()
            .profile_name("default")
            .build(),
    );
    let client = aws_sdk_ec2::Client::new(
        &Config::builder()
            .region(region.clone())
            .credentials_provider(profile)
            .build(),
    );
    println!("Hello, world!");
}
```

This compiles, however the `aws_sdk_ec2::Client` type is not understood by the plugin.

image

And code completion does not work:

image

### PR DESCRIPTION
**Pull request by vlad20012**
Allow files generated by the `build.rs` buildscripts up to 8 megabytes in size. Previously is was necessary to adjust the limit manually using `idea.max.intellisense.filesize` property.

I've tested this with `chrono-tz` crate that generates large `timezones.rs` file in IDEA with default `-Xmx768M`.

Cargo.toml:
```toml
[dependencies]
chrono-tz = "0.6.1"
```

main.rs:
```rust
use chrono_tz::Tz;

fn main() {
    let a = Tz::US__Pacific;
}
```

| Before | After |
| - | - |
| image | image |

It works file until I open the large file. Then I see "Low memory" notification and the IDE starts lagging :(

Fixes #8201, fixes #6134

changelog: Allow files generated by the `build.rs` buildscripts up to 8 megabytes in size. Previously is was necessary to adjust the limit manually using `idea.max.intellisense.filesize` property.

### CODE REVIEW

**Review by Undin — COMMENTED** [review]
Actually, it doesn't fix #8201 since it's not about generated rust files

**Comment by Undin** [comment]
And maybe add test for this?

File: src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt [hunk 1]
```diff
@@ -615,14 +615,14 @@ private class PackageImpl(
 
     override val dependencies: MutableList<DependencyImpl> = ArrayList()
 
-    override val outDir: VirtualFile? by CachedVirtualFile(outDirUrl)
+    override val outDir: VirtualFile? get() = outDirFile?.takeIf { it.isValid }
```
[the remark this answers is not in the document]
Comment (vlad20012) [reply src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt:618]: A user can delete `target` directory, isn't it? But then, in order to generate these files again, the user have to invoke cargo refresh that also re-creates the workspace model

[the remark this answers is not in the document]
Comment (Undin) [reply src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt:618]: Or generated files will be created by manual compilation/external linter

[the remark this answers is not in the document]
Comment (vlad20012) [reply src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspace.kt:618]: Hmm, in this case size limit bump will not be applied

File: src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspaceData.kt [hunk 2]
```diff
@@ -5,6 +5,7 @@
 
 package org.rust.cargo.project.workspace
 
+import com.intellij.openapi.vfs.VirtualFile
```
Comment (Undin) [inline src/main/kotlin/org/rust/cargo/project/workspace/CargoWorkspaceData.kt:8]: unused

**Review by Undin — APPROVED** [review]
LGTM

**Comment by vlad20012** [comment]
bors r=undin

**Pull request merged**